com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/consistent.py

```diff
@@ -0,0 +1,159 @@
+            drain/checkpoint cycle of the consistent region is triggered.
+        period: The trigger period.  If the trigger is :py:const:`~ConsistentRegionConfig.Trigger.PERIODIC`, this must 
+            be specified, otherwise it may not be specfied.  This may be 
+            either a :py:class:`datetime.timedelta` value or the number of 
+            seconds as a `float`.
+        drainTimeout: Indicates the maximum time in seconds that the drain 
+            and checkpoint of the region is allotted to finish processing. 
+            If the process takes longer than the specified time, a failure 
+            is reported and the region is reset to the point of the 
+            previously successfully established consistent state. The value
+            must be specified as either a 
+            :py:class:`datetime.timedelta` value or the number of seconds 
+            as a `float`.  If not specified, the default value is 180 
+            seconds.
+        resetTimeout: Indicates the maximum time in seconds that the reset
+            of the region is allotted to finish processing. If the process
+            takes longer than the specified time, a failure is reported and
+            another reset of the region is attempted.  The value must be
+            specified  as either a :py:class:`datetime.timedelta` value or 
+            the number of seconds as a `float`.  If not specified, the 
+            default value is 180 seconds.
+        maxConsecutiveAttempts(int): Indicates the maximum number of 
+            consecutive attempts to reset a consistent region. After a 
+            failure, if the maximum number of attempts is reached, the 
+            region stops processing new tuples. After the maximum number 
+            of consecutive attempts is reached, a region can be reset only 
+            with manual intervention or with a program with a call to a 
+            method in the consistent region controller.  This must be an 
+            integer value between 1 and 2147483647, inclusive.  If not 
+            specified, the default value is 5.
+
+    Example:
+    ::
+        # set source to be a the start of an operator driven consistent region
+        # with a drain timeout of five seconds and a reset timeout of twenty seconds.
+        source.set_consistent(ConsistentRegionConfig.operatorDriven(drainTimeout=5, resetTimeout=20)
+
+    .. seealso:: :py:meth:`~streamsx.topology.topology.Stream.set_consistent`
+
+    """
+
+    class Trigger(Enum):
+        """
+        Defines how the drain-checkpoint cycle of a consistent region is triggered.
+        """
+        OPERATOR_DRIVEN = 1
+        """
+        Region is triggered by the start operator.
+        """
+        PERIODIC = 2
+        """
+        Region is triggered periodically.
+        """
+
+    def __init__(self, trigger=None, period=None, drainTimeout=None, resetTimeout=None, maxConsecutiveAttempts=None):
```

Comment:
What does trigger==None mean?

Reply:
Trigger must be specified.  There is no default. I added a check to ensure that the trigger is n not None.